browser/js/common/directives/navbar/navbar.js

```diff
@@ -1,47 +1,71 @@
-app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state) {
+app.directive('navbar', function ($rootScope, AuthService, AUTH_EVENTS, $state, $window) {
 
-    return {
-        restrict: 'E',
-        scope: {},
-        templateUrl: 'js/common/directives/navbar/navbar.html',
+	return {
+		restrict: 'E',
+		scope: {},
+		templateUrl: 'js/common/directives/navbar/navbar.html',
 
-        link: function (scope) {
+		link: function (scope) {
+			var temp = 0;
+			scope.goToSection = function(val, state) {
+				if (val !== 'null') {
```

Comment:
so you can just do `if (!val)`, as `undefined` and `null` are considered '[falsy](https://www.sitepoint.com/javascript-truthy-falsy/)' values in JS. i would also change the variable names to be more explicit (e.g., `elementToScrollTo` instead of `val` and `stateToGoTo` instead of `state`. idk, just a small preference.